src/Compilers/CSharp/Portable/Symbols/Synthesized/ReadOnlyListType/SynthesizedReadOnlyListTypeSymbol.cs

```diff
@@ -0,0 +1,288 @@
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iCollectionT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_ICollection_T__Add)), (f, m) => generateNotSupportedException(f, m)));
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iCollectionT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_ICollection_T__Clear)), (f, m) => generateNotSupportedException(f, m)));
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iCollectionT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_ICollection_T__Contains)), (f, m) => generateNotSupportedException(f, m))); // PROTOTYPE: Should be supported.
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iCollectionT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_ICollection_T__CopyTo)), (f, m) => generateNotSupportedException(f, m))); // PROTOTYPE: Should be supported.
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iCollectionT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_ICollection_T__Remove)), (f, m) => generateNotSupportedException(f, m)));
+            addProperty(membersBuilder, new SynthesizedReadOnlyListIndexerProperty(this, (PropertySymbol)getMember(iListT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IList_T__Item)), (f, m) => generateIndexer(f, m), (f, m) => generateNotSupportedException(f, m)));
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iListT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IList_T__IndexOf)), (f, m) => generateNotSupportedException(f, m))); // PROTOTYPE: Should be supported.
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iListT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IList_T__Insert)), (f, m) => generateNotSupportedException(f, m)));
+            membersBuilder.Add(new SynthesizedReadOnlyListMethod(this, (MethodSymbol)getMember(iListT, compilation.GetWellKnownTypeMember(WellKnownMember.System_Collections_Generic_IList_T__RemoveAt)), (f, m) => generateNotSupportedException(f, m)));
+
+            _members = membersBuilder.ToImmutableAndFree();
+
+            // PROTOTYPE: Null-enable.
+#nullable disable
+            static Symbol getMember(NamedTypeSymbol container, Symbol? symbol)
+            {
+                return symbol is null ?
+                    null :
+                    symbol.SymbolAsMember(container);
```

Comment:
```suggestion
                return symbol is null
                    ? null
                    : symbol.SymbolAsMember(container);
```

Aligns punctuation, and makes it clearer taht each line represents one of hte child clauses.